### DISCUSSION TITLE
遇到namespace

### DISCUSSION THREAD

**Discussion by ipanda555** [Yang-Xijie/yang-xijie.github.io#16] [Cpp in VSCode on macOS]
在学习C++语法时，学到命名空间时，发现在hpp文件里，只要加入除了声明函数之外的东西就会报错，比如：

```cpp
namespace A{
  void goAtk(){
    cout <<"+100"<< endl;  //指定在A命名空间内的这个goAtk（）操作
  }
}
```

这样会报错，我尝试去写：

```cpp
#include <iostream>
using namespace std;

void goAtk() {
  cout << "hello" << endl;
}
```

同样会报错，但是只要把除了函数定义之外的都删除，他就不回报错，这是VScode的什么问题吗？之前的VS就不会报错

**Comment by Yang-Xijie** [comment]
（帮你格式化了一下代码）
`h`文件、`hpp`文件都是头文件，一般来说只放声明…我感觉报错是正常的
另外，提问的时候你可以把报错信息粘贴或者截图放过来，这样别人也知道到底报了什么错，是警告还是提示
`VSCode`报错一般都是`C/C++`这个插件报的，可以看看这个插件的设置，根据自己的需求调整
